src/regdesk/printer_manager.js

```diff
@@ -0,0 +1,66 @@
+import { LabelEpl, LP2844 } from 'WebZLP/src/LP2844'
+
+export class PrinterManager {
+    #devices = [];
+    #printers = [];
+
+    primaryPrinter = 0;
+    secondaryPrinter = 1;
+
+    #printerType = LP2844;
+
+    constructor() {
+    }
+
+    get printers() { return this.#printers; }
+
+    async printerConnect(e) {
```

Comment:
Usually methods are verbs, so "connectPrinter", unless these are meant to be event driven in which case maybe "onPrinterConnect"? I'm not super picky admittedly.

Edit: Ahh, I think I see now. This is both in response to a printer connecting, and also does the connection. That feels kinda funny, but I can't think of a better way to do it. Maybe there needs to be a distinction between "onUsbDeviceConnected" (like plugged in) and "nowConnectThePrinter" (actually initialize the thing).

Reply:
Internet says [handleConnect](https://stackoverflow.com/a/60052668/1309423) would be best, I'll roll with that.

The lifecycle is dependent on whether the user has allowed this printer in the past. 

The normal lifecycle is that we ask Chrome to go looking for a USB device filtered by a manufacturer's ID (Zebra, in our case) and the user gets a little popup asking them to select what device they'd like the page to talk to. Once the user selects a device that's what we get returned from the `requestDevice` method.

If they have previously paired the printer (and the printer doesn't have any weirdness going on like a missing serial number) then Chrome can skip the pairing permission process and just fire off the `connect` event to tell us the device reappeared for us to play with.

Since we've finished the `requestDevice` pairing workflow and are now holding a device just like if the `connect` event fired, might as well just turn around and hand it off. It's the same codepath.